packages/suite/src/actions/suite/metadataProviderActions.ts

```diff
@@ -105,9 +108,13 @@ export const disconnectProvider =
         removeMetadata?: boolean;
     }) =>
     async (dispatch: Dispatch) => {
-        Object.values(fetchIntervals).forEach((deviceState, num) => {
-            clearInterval(num);
-            delete fetchIntervals[deviceState];
+        // @ts-expect-error: stupid forEach type
+        Object.keys(fetchIntervals).forEach((id: FetchIntervalTrackingId) => {
```

Comment:
Better than turning the check of IMO:

```suggestion
        (Object.keys(fetchIntervals) as FetchIntervalTrackingId[]).forEach((id: FetchIntervalTrackingId) => {
```

Reply:
2d8f7ea575b1f7104f5c7213d7ffd1825698a501